src/developer/web-api/metadata.md

```diff
@@ -1482,6 +1482,226 @@ following payload to change the style:
+|---------------------------------|-----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Event                           | eventDataValues | action based on merge strategy (DISCARD / LAST_UPDATED). DISCARD will delete all source event data values. LAST_UPDATED will use the event data value which was last updated, when more than one exists. |
+| DataValue                       | dataElement     | action based on merge strategy (DISCARD / LAST_UPDATED). DISCARD will delete all source data values. LAST_UPDATED will use the data value which was last updated, when more than one exists.             |
+| TrackedEntityDataValueChangeLog |                 | deleted if sources are being deleted, otherwise no action.                                                                                                                                               |
+| DataValueAudit                  |                 | deleted if sources are being deleted, otherwise no action.                                                                                                                                               |
+
+
+#### Validation
+
+The following constraints and error codes apply.
+
+Table: Constraints and error codes
+
+| Error code | Description                                                                                                                                 |
+|------------|---------------------------------------------------------------------------------------------------------------------------------------------|
+| E1550      | At least one source data element must be specified                                                                                          |
+| E1551      | Target data element must be specified                                                                                                       |
+| E1552      | Target data element cannot be a source indicator                                                                                            |
+| E1553      | Source/Target data element does not exist: `{uid}`                                                                                          |
+| E1554      | All source ValueTypes must match target ValueType: `ValueType`. Other ValueTypes found: `ValueType`                                         |
+| E1555      | All source DataElementDomains must match target DataElementDomain: `DataElementDomain`. Other DataElementDomains found: `DataElementDomain` |
+| E1556      | dataMergeStrategy field must be specified. With value `DISCARD` or `LAST_UPDATED`                                                           |
+
+#### Database constraints
+If a database table has a unique key constraint, allowing only 1 unique data element as part of the constraint, then the merge will fail.  
```

Comment:
This sounds unclear to me. Why would there be one or not be one? Who adds the constraint? Should one expect a constraint to be present? What is the situation where we get a collision?

I assume you mean this referring to the table that follows. I think we better state this as a given. Something like "There are unique constraints in place that can prevent a successful merge". When would this be a conflict compare to when does the merge work? I don't quite get the condition from this. What quality of the source or target is important here?

Reply:
this information is all under the section `Database constraints`. So it is already logically grouped together. It outlines, in order:
- warning that DB constraints exist
- why they can occur
- when/where they can occur

I can update to say that these constraints are set by DHIS2 core team (this seems to be a very obvious detail though 😄 )
I can also update to say why a DB constraint might exist, which are basically domain constraints set by DHIS2.
And I can use your proposed alternative suggestion as well.

Reply:
The intro just sounded too loose where this might be up to users or operators/admins when I assume it is more correct to assume this is simply part of the software package.

Reply:
gotcha 👍